src/DynamoDbModel.php

```diff
@@ -86,11 +86,38 @@ protected function setupDynamoDb()
 
     public function save(array $options = [])
     {
-        if (!$this->getKey()) {
-            $this->fireModelEvent('creating');
+        $create = !$this->exists;
+        if ($create) {
+            if ($this->fireModelEvent('creating')  === false) {
+                return false;
+            }
+        } else {
+            if ($this->fireModelEvent('updating') === false) {
+                return false;
+            }
+        }
```

Comment:
The shorter conditionals are definitely OK - however, at the bottom we still need to test against $create for the "wasRecentlyCreated" as an update isn't "recently created".  Working on updating this on my branch now.